common/src/main/java/edu/snu/nemo/common/ir/vertex/MetricCollectionBarrierVertex.java

```diff
@@ -71,21 +74,28 @@ public void setDAGSnapshot(final DAG<IRVertex, IREdge> dag) {
 
   /**
    * Method for accumulating metrics in the vertex.
-   * @param key metric key, e.g. ID of the partition.
-   * @param values metric values, e.g. the block size information of the partition data.
+   * @param metric the block size information of the partition.
    */
-  public void accumulateMetric(final String key, final List<T> values) {
-    metricData.putIfAbsent(key, values);
+  public void updateMetricData(final T metric) {
+    metricData = metric;
```

Comment:
The comment makes it seem like it's narrowing down the functionality just for keeping track of block sizes. Would there be a particular reason? Also, what this code does is 'set' instead of 'update'. I think this makes everything manual, and leaves the implementation of the metric accumulation to the part using the APIs, instead of factoring out the repetitive step in this class. I would like to hear a bit more about the rational behind this change.

Reply:
In current design, we can only either conduct on-demand metric aggregation in RuntimeMaster manually like this, or make MetricCollectionBarrierVertex aggregate metric on-demand(in this case, adding partition sizes as message arrives) by explicitly typecast metric value to `Long`. Either way is ad hoc.

I think I should resolve this issue in NEMO-96, by replacing MetricCollectionBarrierVertex to AggregationBarrierVertex that accepts user-defined aggregation function and does on-demand metric aggregation.